### ISSUE TITLE
Handle exceptions better

### ISSUE THREAD

**Issue by jethrocarr** [jethrocarr/pupistry#4]
Currently Pupistry is a bit of a mix of using return codes and using exceptions. We should fix it to properly do exception handling everywhere and dump the use of return codes for success/failure.

**Comment by mexisme** [comment]
As a first step, you may want to create an Exception object that encapsulates the return codes, and implement helper methods that allow it to be used in-place throughout your code.
e.g.

```
def ==(other)
  other == @err_code
end
```

I have a feeling something similar may have already been written in one of the IO classes that you could sub-class + extend effectively.

**Comment by jethrocarr** [comment]
Good idea - got to do something, I hacked in using return codes everywhere and every time I look at the code it now fills me with unlimited amounts of shame and regret.
